tests/test_incremental_sync.py

```diff
@@ -0,0 +1,58 @@
+from LSP.plugin.core.typing import Generator, Tuple
+from LSP.plugin.core.url import filename_to_uri
+from setup import TextDocumentTestCase
+from setup import YieldPromise
+from test_single_document import TEST_FILE_PATH
+
+
+class TestIncrementalSync(TextDocumentTestCase):
+
+    def setUp(self) -> Generator:
+        yield from super().setUp()
+        self.maxDiff = None
+
+    def verify(self, *args: Tuple[int, int, int, int, int, str]) -> Generator:
+        promise = YieldPromise()
+        yield from self.await_message("textDocument/didChange", promise)
+        content_changes = []
+        for arg in args:
+            content_changes.append({
+                'rangeLength': arg[0],
+                'range': {
+                    'start': {'line': arg[1], 'character': arg[2]},
+                    'end': {'line': arg[3], 'character': arg[4]}},
+                'text': arg[5]})
+        self.assertEqual(promise.result(), {
+            'contentChanges': content_changes,
+            'textDocument': {'version': self.view.change_count(), 'uri': filename_to_uri(TEST_FILE_PATH)}
+        })
+
+    def test_single_insert(self) -> Generator:
+        self.insert_characters("a")
+        yield from self.verify(
+            (0, 0, 0, 0, 0, 'a')
+        )
+
+    def test_neighboring_inserts(self) -> Generator:
+        self.insert_characters("a")
+        yield 40
```

Comment:
I suppose just `yield` would be as effective?
That's assuming that it's enough to break out of the main thread for a moment.

Reply:
Turns out no yield is needed at all.